### PR TITLE
[compiler-rt] Fix build for android

### PR DESCRIPTION
**Pull request by chestnykh**
Fixes
https://lab.llvm.org/buildbot/#/builders/186/builds/1047

### CODE REVIEW

File: compiler-rt/lib/sanitizer_common/sanitizer_linux.cpp [hunk 1]
```diff
@@ -2154,7 +2154,8 @@ bool SignalContext::IsTrueFaultingAddress() const {
 UNUSED
 static const char *RegNumToRegName(int reg) {
   switch (reg) {
-#  if SANITIZER_LINUX && SANITIZER_GLIBC || SANITIZER_NETBSD
+#  if (SANITIZER_LINUX && SANITIZER_GLIBC && !SANITIZER_ANDROID) || \
```
Comment (devnexen) [inline compiler-rt/lib/sanitizer_common/sanitizer_linux.cpp:2157]: I m surprised you need to indicate android since there is glibc involved but lgtm..

Comment (chestnykh) [reply compiler-rt/lib/sanitizer_common/sanitizer_linux.cpp:2157]: I see a lot of places inside compiler_rt where we have `SANITIZER_LINUX && !SANITIZER_ANDROID`

Comment (chestnykh) [reply compiler-rt/lib/sanitizer_common/sanitizer_linux.cpp:2157]: i don't know actually if android uses glibc but it seems that yes

Comment (chestnykh) [reply compiler-rt/lib/sanitizer_common/sanitizer_linux.cpp:2157]: I hope this is the last patch with adjustments of preprocessor directives

Comment (devnexen) [reply compiler-rt/lib/sanitizer_common/sanitizer_linux.cpp:2157]: hmmm strange still because android libc is bionic :)

Comment (chestnykh) [reply compiler-rt/lib/sanitizer_common/sanitizer_linux.cpp:2157]: Of course this is strange. __GLIBC__ is defined somewhere

Comment (devnexen) [reply compiler-rt/lib/sanitizer_common/sanitizer_linux.cpp:2157]: > Of course this is strange. **GLIBC** is defined somewhere

See sanitizer_platform.h comment line 21. Anyhow, feel free to merge or wait @vitalybuka eventually.

Comment (chestnykh) [reply compiler-rt/lib/sanitizer_common/sanitizer_linux.cpp:2157]: Maybe, but i cannot quicky deploy android env and check this precisely...
So i think we can leave this patch unchanged to make buildbots green

Comment (chestnykh) [reply compiler-rt/lib/sanitizer_common/sanitizer_linux.cpp:2157]: Lets wait @vitalybuka

Comment (chestnykh) [reply compiler-rt/lib/sanitizer_common/sanitizer_linux.cpp:2157]: @vitalybuka tell us please what is the correct combination of macroses here :)

Comment (vitalybuka) [reply compiler-rt/lib/sanitizer_common/sanitizer_linux.cpp:2157]: Android does not use SANITIZER_GLIBC

Comment (vitalybuka) [reply compiler-rt/lib/sanitizer_common/sanitizer_linux.cpp:2157]: On revision from the build error, there is no
`if SANITIZER_LINUX && SANITIZER_GLIBC || SANITIZER_NETBSD` in `RegNumToRegName`

**Review by vitalybuka — CHANGES_REQUESTED** [review]
"Request changes" to hide from my review dashboard.
I suspect it's not needed.

If it's needed, please explain how to reproduce the issue, I can try on that bot.
And please "re-request review" if it's still relevant.

Please

**Pull request closed without merging**